cmd/README.md

```diff
@@ -0,0 +1,655 @@
+# Precompile Generation Tutorial
+A stateful precompile allows us to add more functionality and customization to the EVM. A stateful precompile builds on a precompile in that it adds state access. This means our precompile can manage state as well as interacting with EVM state. 
+
+A stateful precompile follows this interface. 
+``` go
+// StatefulPrecompiledContract is the interface for executing a precompiled contract
+type StatefulPrecompiledContract interface {
+	// Run executes the precompiled contract.
+	Run(accessibleState PrecompileAccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error)
+}
+
+```
+
+It looks like we have to create an actual precompile, an interface for the precompile, a contract that interacts with the precompile, tests for that contract, and small modifications throughout the EVM to enable the precompile. This seems like a lot of work. Luckily, we have the precompile generation tool! 
+
+This tool does a bulk of the precompile creation work in one command. In this tutorial, we will be walking through how to create a working precompile from scratch. 
+
+### Assumption of Knowledge
+Before building a stateful precompile, you will want to make sure that you have an understanding of the EVM, precompiles, and stateful precompiles. 
+Here are some resources to get started put together. 
+
+- [The Ethereum Virtual Machine](https://github.com/ethereumbook/ethereumbook/blob/develop/13evm.asciidoc)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Deconstructing a Smart Contract](https://blog.openzeppelin.com/deconstructing-a-solidity-contract-part-i-introduction-832efd2d7737/)
+- [Layout of State Variables in Storage](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_storage.html)
+- [Layout in Memory](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_memory.html)
+- [Layout of Call Data](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_calldata.html)
+- [Contract ABI Specification](https://docs.soliditylang.org/en/v0.8.10/abi-spec.html)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+
+ 
+## Tutorial
+
+Let's start by creating the Solidity interface that we want our precompile to implement. We will place this file in `./contract-examples/contracts`
```

Comment:
Can we add more detail here so that it's clear what the interface will do. We should say something about implementing a Hello World interface before laying out the solidity.